string obj_name
int64 number